src/decoder/stateless/av1.rs

```diff
@@ -176,14 +184,19 @@ where
                 handle: ref_frame.clone(),
             });
         } else if let Some(sequence) = &self.codec.sequence {
-            let backend_picture = self.backend.new_picture(
-                sequence,
+            let mut backend_picture = self.backend.new_picture(
                 &frame_header,
                 timestamp,
-                &self.codec.reference_frames,
                 self.codec.highest_spatial_layer,
             )?;
 
+            self.backend.begin_picture(
+                &mut backend_picture,
+                sequence,
+                &frame_header,
+                &self.codec.reference_frames,
+            )?;
```

Comment:
Just to clarify. Since the code is calling `begin_picture` shortly after `new_picture`, thus there is no change when the picture is allocated?

Reply:
Correct. The goal is just to add a point where the backend can tell that it cannot go ahead due to lack of resources.

Initially everything was done in the former `new_picture`, but I thought it would be cleaner to separate allocation and initialization in the hope of having all codecs converge towards a common `new_picture` method. We are pretty close I think.

For consistency I should probably do the same split for VP8...

Reply:
Thanks for clarification!